amethyst_assets/src/reload.rs

```diff
@@ -1,18 +1,203 @@
-//! MODULE DISABLED, will be reused later!
+//! Defines the `Reload` trait.
 
-use Source;
+use std::sync::Arc;
+use std::time::Instant;
 
-pub trait Reload {
-    fn needs_reload(&self, source: &Source) -> bool;
+use {Asset, BoxedErr, Format, FormatValue, Source};
+
+/// An ECS resource which allows to configure hot reloading.
+///
+/// ## Examples
+///
+/// ```
+/// # extern crate amethyst_assets;
+/// # extern crate specs;
+/// #
+/// # use amethyst_assets::HotReloadStrategy;
+/// # use specs::World;
+/// #
+/// # fn main() {
+/// let mut world = World::new();
+/// // Assets will be reloaded every two seconds (in case they changed)
+/// world.add_resource(HotReloadStrategy::every(2));
+/// # }
+/// ```
+pub struct HotReloadStrategy {
+    inner: HotReloadStrategyInner,
+}
+
+impl HotReloadStrategy {
+    /// Causes hot reloads every `n` seconds.
+    pub fn every(n: u8) -> Self {
+        HotReloadStrategy {
+            inner: HotReloadStrategyInner::Every {
+                interval: n,
+                start: Instant::now(),
+            },
+        }
+    }
+
+    /// No periodical hot-reloading is performed.
+    /// Instead, you can `trigger` it to check for changed assets
+    /// in one specific frame.
+    pub fn never() -> Self {
+        HotReloadStrategy {
+            inner: HotReloadStrategyInner::Never,
+        }
+    }
+
+    /// The frame after calling this, all changed assets will be reloaded.
+    pub fn trigger(&mut self) {
+        let counter = match self.inner {
+            HotReloadStrategyInner::Trigger { counter } => counter.checked_add(1).unwrap_or(0),
```

Comment:
Just `wrapping_add` instead of `checked_add` and `unwrap_or`

Reply:
That's what I've been searching for, thanks!